compiler/src/iree/compiler/Codegen/TransformStrategies/GPU/AbstractGemmLikeStrategy.cpp

```diff
@@ -89,6 +89,10 @@ void AbstractGemmLikeStrategy::initDefaultValues() {
       pipelineDepth != clPipelineDepth.getDefault().getValue()) {
     cliOptionsSpecified = true;
   }
+
+  // Adjust default reduction tile size for f16.
+  if (reductionTileSize == clReductionTileSize.getDefault().getValue())
```

Comment:
This turns out to be very confusing .. it does override my passing a 16 explicitly.

Reply:
Makes sense, we probably need to revisit the way that these command line arguments are specified because different bit widths will want different defaults. The defaults probably shouldn't be encoded as flag defaults and instead the flags all become opt-in with default "null" values. I can make that change and drop this.

Reply:
I opened #14222 to improve this. It should also help to generalize to other targets/matmuls that might want/require different defaults.